marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/entitySearch/getRecordTest.sjs

```diff
@@ -0,0 +1,95 @@
+const entitySearchService = require("/test/suites/data-hub/5/data-services/lib/entitySearchService.sjs");
+
+function getJsonRecordTest() {
+  const docUri = "/content/jane.json";
+  xdmp.invokeFunction(function() {
+    declareUpdate();
+    xdmp.documentSetMetadata("/content/jane.json",
+        {
+          "datahubCreatedByStep": "map-step",
+          "datahubCreatedInFlow": "CurateCustomerJSON",
+          "datahubCreatedOn": "2020-10-08T15:14:28.772612-07:00"
+        });
+  });
+
+  const record = entitySearchService.getRecord(docUri);
+  const expectedMetadata = JSON.stringify({
+    "datahubCreatedByStep": "map-step",
+    "datahubCreatedInFlow": "CurateCustomerJSON",
+    "datahubCreatedOn": "2020-10-08T15:14:28.772612-07:00"
+  });
+
+  return[
+    test.assertEqual(fn.doc(docUri), record.data),
+    test.assertEqual(expectedMetadata, JSON.stringify(record.recordMetadata)),
+    test.assertEqual(true, record.isHubEntityInstance),
+    test.assertEqual("json", record.recordType)
+  ];
+}
+
+function getXmlRecordTest() {
+  const docUri = "/content/jane.xml";
+  xdmp.invokeFunction(function() {
+    declareUpdate();
+    xdmp.documentSetMetadata("/content/jane.xml",
+        {
+          "datahubCreatedByStep": "map-step",
+          "datahubCreatedInFlow": "CurateCustomerXML",
+          "datahubCreatedOn": "2020-10-08T15:14:28.772612-07:00"
+        });
+  });
+
+  const record = entitySearchService.getRecord(docUri);
+  const expectedMetadata = JSON.stringify({
+    "datahubCreatedByStep": "map-step",
+    "datahubCreatedInFlow": "CurateCustomerXML",
+    "datahubCreatedOn": "2020-10-08T15:14:28.772612-07:00"
+  });
+
+  return[
+    test.assertEqual(fn.doc(docUri), record.data),
+    test.assertEqual(expectedMetadata, JSON.stringify(record.recordMetadata)),
+    test.assertEqual(true, record.isHubEntityInstance),
+    test.assertEqual("xml", record.recordType)
+  ];
+}
+
+function getTextRecordTest() {
+  const docUri = "/content/jane.txt";
+  xdmp.invokeFunction(function() {
+    declareUpdate();
+    xdmp.documentSetMetadata("/content/jane.txt",
+        {
+          "datahubCreatedByStep": "ingest-step",
+          "datahubCreatedInFlow": "ingestText",
+          "datahubCreatedOn": "2020-10-08T15:14:28.772612-07:00"
+        });
+  });
+
+  const record = entitySearchService.getRecord(docUri);
+  const expectedMetadata = JSON.stringify({
+    "datahubCreatedByStep": "ingest-step",
+    "datahubCreatedInFlow": "ingestText",
+    "datahubCreatedOn": "2020-10-08T15:14:28.772612-07:00"
+  });
+
+  return[
+    test.assertEqual(fn.doc(docUri), record.data),
+    test.assertEqual(expectedMetadata, JSON.stringify(record.recordMetadata)),
+    test.assertEqual(false, record.isHubEntityInstance),
+    test.assertEqual("text", record.recordType)
+  ];
+}
+
+function getNonExistentRecordTest() {
+  const docUri = "/content/nonExistentRecord.json";
+  const record = entitySearchService.getRecord(docUri);
+  console.log(record);
+}
+
+[]
```

Comment:
Should these be uncommented now? It doesn't appear that anything is being tested here.

Reply:
Thank you for noticing it. I updated it